Std/Data/List/Perm.lean

```diff
@@ -0,0 +1,1029 @@
+/-
+Copyright (c) 2015 Microsoft Corporation. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Leonardo de Moura, Jeremy Avigad, Mario Carneiro
+
+! This file was ported from Lean 3 source module data.list.perm
+! leanprover-community/mathlib commit 7b78d1776212a91ecc94cf601f83bdcc46b04213
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Std.Data.List.Lemmas
+import Std.Data.List.Count
+import Std.Data.List.Pairwise
+
+/-!
+# List Permutations
+
+This file introduces the `List.Perm` relation, which is true if two lists are permutations of one
+another.
+
+## Notation
+
+The notation `~` is used for permutation equivalence.
+-/
+
+
+open Nat
+
+universe uu vv
+
+namespace List
+
+variable {α : Type uu} {β : Type vv} {l₁ l₂ : List α}
+
+/-- `Perm l₁ l₂` or `l₁ ~ l₂` asserts that `l₁` and `l₂` are permutations
+  of each other. This is defined by induction using pairwise swaps. -/
+inductive Perm : List α → List α → Prop
+  /-- `[] ~ []` -/
+  | nil : Perm [] []
+  /-- `l₁ ~ l₂ → x::l₁ ~ x::l₂` -/
+  | cons (x : α) {l₁ l₂ : List α} : Perm l₁ l₂ → Perm (x :: l₁) (x :: l₂)
+  /-- `x::y::l ~ y::x::l` -/
+  | swap (x y : α) (l : List α) : Perm (y :: x :: l) (x :: y :: l)
+  /-- `Perm` is transitive. -/
+  | trans {l₁ l₂ l₃ : List α} : Perm l₁ l₂ → Perm l₂ l₃ → Perm l₁ l₃
+
+
+open Perm (swap)
+
+/-- `Perm l₁ l₂` or `l₁ ~ l₂` asserts that `l₁` and `l₂` are permutations
+  of each other. This is defined by induction using pairwise swaps. -/
+infixl:50 " ~ " => Perm
+
+@[simp]
+protected theorem Perm.refl : ∀ l : List α, l ~ l
+  | [] => Perm.nil
+  | x :: xs => (Perm.refl xs).cons x
+
+
+-- Porting note: used rec_on in mathlib3; lean4 eqn compiler still doesn't like it
+protected theorem Perm.symm {l₁ l₂ : List α} (p : l₁ ~ l₂) : l₂ ~ l₁ :=
+  p.rec
+    .nil
+    (fun x _ _ _ r₁ => .cons x r₁)
+    (fun x y l => .swap y x l)
+    (fun _ _ r₁ r₂ => .trans r₂ r₁)
+
```

Comment:
Style-wise, is there any reason for double newlines between theorems? Afaict one-line is the norm (just because more fits on the screen then).

Reply:
I assume it is a side effect of removing the `#align` lines. I agree the double newlines should be removed